### PR TITLE
Add function to load secret/password from file for security

### PR DESCRIPTION
**Pull request by dabde**
For security reasons, we don't allow passwords/credentials in environment variables. So I added a small addition to load the SMTP_PASSWORD from a file.

### CODE REVIEW

**Comment by juanluisbaptiste** [comment]
Hi @dabde , thanks for the PR, I will test it out first as soon as I can.

File: run.sh [hunk 1]
```diff
@@ -13,6 +13,9 @@ function add_config_value() {
  postconf -e "${key} = ${value}"
 }
 
+# Read password from file to aviod unsecure env variables
+[ -n "${SMTP_PASSWORD_FILE}" ] && read SMTP_PASSWORD < ${SMTP_PASSWORD_FILE}
```
Comment (juanluisbaptiste) [inline run.sh:17]: You should check if the file exists, not only if the env variable is not null.

File: README.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ The following env variables need to be passed to the container:
 * `SMTP_PORT` (Optional, Default value: 587) Port address of the SMTP server to use.
 * `SMTP_USERNAME` Username to authenticate with.
 * `SMTP_PASSWORD` Password of the SMTP user.
+* (optional) `SMTP_PASSWORD_FILE` File Ref to load SMTP_PASSWORD, to aviod passwords in enviorment variables.
```
Comment (juanluisbaptiste) [inline README.md:45]: Please fix typos and rephrase this to make it more clear that this new env var contains the path to a file that contains the smtp password instead of setting it on SMTP_PASSWORD variable.

**Comment by juanluisbaptiste** [comment]
Hi @dabde ,

First of all I'm sory for the delay on reviewing this. Finally I had some time to take a look to it, please take a look at my inline comments.

**Comment by dabde** [comment]
hi @juanluisbaptiste ,

thanks for the feedback, hope my changes solved your comments.

**Comment by juanluisbaptiste** [comment]
@dabde , did you test your code ? is not working:

```
postfix_1  | SMTP_PASSWORD is not set
postfix_1  | + '[' -n /run/passwd ']'
postfix_1  | + '[' -f /run/passwd ']'
postfix_1  | + read SMTP_PASSWORD
postfix_1  | + '[' -z smtp.sendgrid.net ']'
postfix_1  | + '[' -z apikey ']'
postfix_1  | + '[' -z '' ']'
postfix_1  | + echo 'SMTP_PASSWORD is not set'
postfix_1  | + exit 1

AFAIK the read function is used to read a value _from_ stdin into a variable, I don't understand how you want to use it to read a file instead.
```

File: README.md [hunk 3]
```diff
@@ -50,6 +50,8 @@ The following env variable(s) are optional.
 * `SMTP_NETWORKS` Setting this will allow you to add additional, comma seperated, subnets to use the relay. Used like
     -e SMTP_NETWORKS='xxx.xxx.xxx.xxx/xx,xxx.xxx.xxx.xxx/xx'
 
+* `SMTP_PASSWORD_FILE` Setting this to a mounted file containing the password, to avoid passwords in env variables.
```
Comment (juanluisbaptiste) [inline README.md:53]: It should be clear that this variable should point to a file inside the container, an example could help here. It also needs to be added and documented in the .env.example file.

**Comment by dabde** [comment]
@juanluisbaptiste sry, not the `read` was the issue. With the condition I added brackets, was not testing the full code, only if the condition is working fine. But with the brackets, the sub-shell can't set vars for the script :facepalm: .

changed the code and tested completely.
```
bash-5.0# cat /etc/postfix/sasl_passwd
[smtp.office365.com]:25 XXXXXXXXXXX:12123123
bash-5.0# env | grep "FILE"
SMTP_PASSWORD_FILE=/sec/pass
bash-5.0# cat /sec/pass
12123123
```

and yes, `read` is used for stdin, but you can pipe to stdin in this case. And in this way, I only read the first line, avoid accidental line breaks by user. If you have another fitting option, I'm open to change this.
```
#!/bin/bash

set -x

read SMTP_PASSWORD < ${SMTP_PASSWORD_FILE}
echo ${SMTP_PASSWORD}

SMTP_PASSWORD=$(cat ${SMTP_PASSWORD_FILE})
echo ${SMTP_PASSWORD}

SMTP_PASSWORD=$(<${SMTP_PASSWORD_FILE})
echo ${SMTP_PASSWORD}
```
Output
```
bash ./test.sh
+ read SMTP_PASSWORD
+ echo a
a
++ cat ./file
+ SMTP_PASSWORD='a
b
c'
+ echo a b c
a b c
+ SMTP_PASSWORD='a
b
c'
+ echo a b c
a b c

```

File: README.md (hunk 3 above)
Comment (dabde) [reply README.md]: added usage to the readme and added var to .env.example

**Comment by juanluisbaptiste** [comment]
Now I see it working, thanks.

**Comment by Montellese** [comment]
Thanks for this useful functionality.

Unfortunately using `read` to read the password from the file the way it is currently done can result in the error message being dsiplayed even though the password has been properly read and everything is working fine. The problem occurs if the password file doesn't contain a newline. Then `read` will hit EOF and return with exit code `1`. `SMTP_PASSWORD` will still be valid but in the logs it will say `SMTP_PASSWORD_FILE defined, but file not existing, skipping.`.

It's not really a big deal since everything is working fine but it cost me a few hours to figure this out...
